controller/commentsController.js

```diff
@@ -51,3 +51,48 @@ exports.flagComment = async (req, res) => {
     errHandler(error, res);
   }
 };
+
+exports.deleteComment = async (req, res) => {
+  const commentId = req.params.commentId;
+  const userEmail = req.body.commentOwnerEmail;
+  if (!commentId || !userEmail) {
+    return res.status(400).json({
+      status: false,
+      message: "Comment Id and email address required",
+    });
+  }
+  try {
+    const comment = await Comments.findOne({ _id: commentId }).populate(
+      "commentOwner"
+    );
+    if (!comment) {
+      return res.status(400).json({
+        status: false,
+        message: "Comment not found",
+      });
+    }
+    const email = await comment;
+    if (email.commentOwner.email.toLowerCase() == userEmail.toLowerCase()) {
+      const deleting = await Comments.findByIdAndDelete(commentId);
+      if (deleting) {
+        return res.status(200).json({
+          status: true,
+          message: "Comment deleted successfully",
+        });
+      } else {
+        return res.status(400).json({
```

Comment:
Please use the responseHandler from the utils folder.

Also, should this return a status code of 400?

Reply:
what status code should it return. its a response for if the user is not the owner of the comment. so no req goes through

Reply:
You are right. Nevermind that.